core/src/commonMain/kotlin/org/aquila3d/core/math/Quaternion.kt

```diff
@@ -0,0 +1,324 @@
+     * Constructs a new [Quaternion] from the provided [Double] components.
+     */
+    constructor(w: Double, x: Double, y: Double, z: Double) {
+        this.w = w
+        this.x = x
+        this.y = y
+        this.z = z
+    }
+
+    /**
+     * Constructs a new unit [Quaternion] from a provided [Quaternion].
+     */
+    constructor(other: Quaternion) {
+        this.w = other.w
+        this.x = other.x
+        this.y = other.y
+        this.z = other.z
+    }
+
+    /**
+     * Constructs a new unit [Quaternion] from a given rotation about a given axis. The inputs do not need to be
+     * normalized.
+     *
+     * @param angle Angle of ratation in [Radians]
+     * @param axis [Vector3] representing the axis of rotation
+     */
+    constructor(angle: Radians, axis: Vector3) {
+        val s = sin(angle.angle / 2.0)
+        val normAxis = Vector3(axis).normalize().multiply(s)
+        w = cos(angle.angle / 2.0)
+        x = normAxis.x
+        y = normAxis.y
+        z = normAxis.z
+    }
+
+    /**
+     * Constructs a new [Quaternion] by applying the provided Euler rotation angles in order, yaw - pitch - roll.
+     * Note that these Euler angles are defined in GL coordinate space, not the typical rigid body mechanics space.
+     *
+     * @param yaw Rotation about the +Y axis, in [Radians]
+     * @param pitch Rotation about the +X Axis, in [Radians]
+     * @param roll Rotation about the +Z Axis, in [Radians]
+     */
+    constructor(yaw: Radians, pitch: Radians, roll: Radians) {
+        val hYaw = yaw.angle / 2.0
+        val hPitch = pitch.angle / 2.0
+        val hRoll = roll.angle / 2.0
+        val sYaw = sin(hYaw)
+        val cYaw = cos(hYaw)
+        val sPitch = sin(hPitch)
+        val cPitch = cos(hPitch)
+        val sRoll = sin(hRoll)
+        val cRoll = cos(hRoll)
+
+        val cYawcPitch = cYaw * cPitch
+        val sYawsPitch = sYaw * sPitch
+        val cYawsPitch = cYaw * sPitch
+        val sYawcPitch = sYaw * cPitch
+
+        w = cRoll * cYawcPitch + sRoll * sYawsPitch
+        x = cRoll * cYawsPitch + sRoll * sYawcPitch
+        y = cRoll * sYawcPitch - sRoll * cYawsPitch
+        z = sRoll * cYawcPitch - cRoll * sYawsPitch
+    }
+
+    operator fun unaryMinus() {
+        invert()
+    }
+
+    operator fun plus(other: Quaternion): Quaternion {
+        return add(other)
+    }
+
+    operator fun minus(other: Quaternion): Quaternion {
+        return subtract(other)
+    }
+
+    operator fun times(scale: Float): Quaternion {
+        return multiply(scale)
+    }
+
+    operator fun times(scale: Double): Quaternion {
+        return multiply(scale)
+    }
+
+    operator fun times(vector3: Vector3): Quaternion {
+        return multiply(vector3)
+    }
+
+    operator fun times(other: Quaternion): Quaternion {
+        return multiply(other)
+    }
+
+    operator fun div(scale: Float): Quaternion {
+        return divide(scale)
+    }
+
+    operator fun div(scale: Double): Quaternion {
```

Comment:
Why not define the operations here instead of the two names for one method?

Reply:
I'm a fan of verbose method names that read like english. My thinking here was to provide the operators for shorthand or the more verbose names for explicit clarity. If you think it will be a performance problem I am fine dropping the other methods.

Reply:
Removed.